gcloud/gcloud_kms.go

```diff
@@ -192,6 +187,82 @@ func (g *gcloudKmsSecrets) PutSecret(
 	)
 }
 
+func getRSALimit(pk *rsa.PublicKey, hash *hash.Hash) int {
+	return pk.Size() - 2*(*hash).Size() - 2
+}
+
+// decrypt takes in the dek, which is a map of encrypted bytes, in byte array form.
+// It first unmarshal the data into the map. Then iterate through each item sequentially to decrypt.
+// Finally combined them into the decrypted plaintext form.
+func decryptToPlaintext(g *gcloudKmsSecrets, dek []byte) ([]byte, error) {
+	var plaintext []byte
+	var dekMap map[int][]byte
+
+	if err := json.Unmarshal(dek, &dekMap); err != nil {
```

Comment:
We need to make this backward compatible. If this is Unmarshal fails we should try decrypting the whole dek blob using Google KMS API and return if it succeeds.

Reply:
added the logic and test to ensure it works.